### PR TITLE
First draft position reconstruction with LRFs and MC based fit parameters.

### PR DESCRIPTION
**Pull request by a1exndr**

### CODE REVIEW

File: straxbra/plugins.py [hunk 1]
```diff
@@ -368,58 +372,80 @@ def compute(self, peaks):
...
+        return (p_TPC_radius - np.sqrt(inputs[0]**2 + inputs[1]**2))
 
+    ## Radial LRF model
+    ## from: Position Reconstruction in a Dual Phase Xenon Scintillation Detector (https://arxiv.org/abs/1112.1481)
+    @staticmethod
+    def eta(r, A, r0, a, b, alpha):
+        return A * exp( - a * (r/r0) / (1 + (r/r0) ** (1 - alpha)) - b / (1 + (r/r0) ** (- alpha)))
+
+    ## Position dependent LRF values individual PMTs from model
+    def LRF_PMTs(x, y, self):
```
Comment (darrylmasson) [inline straxbra/plugins.py:410]: `self` must be the first argument, also all the hard-coded numbers should be moved into `setup()`. Additionally, please don't abuse the lack of enforced line lengths.

File: straxbra/plugins.py [hunk 2]
```diff
@@ -368,58 +372,80 @@ def compute(self, peaks):
...
+    Position Reconstruction for XeBRA with LRFs
+
+    Status: July 2019
+
+    Position reconstruction for XeBRA following the position reconstruction algorithm of Mercury
+    (employed in the LUX experiment, originally developed for the ZEPLIN-III dark matter experiment)
+    with light response functions (see https://arxiv.org/abs/1710.02752v2 and
+    https://arxiv.org/abs/1112.1481 ).
+    '''
     __version__ = "0.0.1"
```
Comment (darrylmasson) [inline straxbra/plugins.py:385]: Increment, please

File: straxbra/plugins.py [hunk 3]
```diff
@@ -368,58 +372,80 @@ def compute(self, peaks):
...
+        PMT_position_x = np.array([-14.,-28,-14.,14.,28.,14.,0.])      # x-position PMTs in mm in cartesian coordinates
+        PMT_position_y = np.array([-28.,0.,28.,28.,0.,-28.,0.])        # y-position PMTs in mm in cartesian coordinates
+        ## Fit parameters LRFs
+        ## MC driven (R_PTFE = 95 %, T_meshes = 89.770509 %, lambda_LXe = 100 cm);
+        ## To be iteratively determined from data later
+        fitparameters = np.array([np.array([ 0.58534179, 29.89341846, -0.3275816, 4.14715081, 2.61234684]), np.array([ 0.63546614, 28.49525342, -0.19651583, 3.76011493, 2.68672152]), np.array([ 0.58840586, 30.39015033, -0.38248759, 4.27642323, 2.56774842]), np.array([ 0.59111988, 31.38350968, -0.5760228, 4.64122418, 2.51311592]), np.array([ 0.63771524, 28.90663204, -0.26194541, 3.90052756, 2.66510948]), np.array([ 0.59030322, 30.55082687, -0.46183924, 4.39769959, 2.55446814]), np.array([ 0.53114467, 39.20861977, -17.93187819, 20.60397171, 2.27692367])])
+        ## LRF values for individual PMTs
+        LRF_PMTs_array = np.array([(self.eta(np.sqrt((x - PMT_position_x[j-1])**2 + (y - PMT_position_y[j-1])**2), *fitparameters[j-1])) for j in range(1,8)])
```
Comment (darrylmasson) [inline straxbra/plugins.py:419]: The offset by one isn't needed as only the area from the top array is passed through.

**Comment by a1exndr** [comment]
I just saw that you apparently entirely updated the PeakPositions class for the old centroids approach. How am I now supposed to implement my position reconstruction with LRFs, shall I define a new class or overwrite yours?

**Comment by darrylmasson** [comment]
The original version didn't work, and I needed working code to test the processing chain. We'll resolve the merge conflicts when your code is finalized. There is no reason to keep the centroid method.

**Comment by a1exndr** [comment]
Alright, then I'll overwrite it to the extent needed to go from the centroid to the LRF method while trying to keep useful changes you just implemented (such as the updated initially wrong PMT positions in setup()).

File: straxbra/plugins.py (hunk 1 above)
Comment (a1exndr) [reply straxbra/plugins.py:410]: Thanks for the hints, they're taken to heart in the updated version.

File: straxbra/plugins.py (hunk 2 above)
Comment (a1exndr) [reply straxbra/plugins.py:385]: Incremented to `__version__ = "0.0.2"`.

File: straxbra/plugins.py (hunk 3 above)
Comment (a1exndr) [reply straxbra/plugins.py:419]: Right, it also worked like that, the shift in indices was only an ugly relic from the original code using pandas dataframes before being boiled down to the minimalist version it is now. Corrected it now and also accounted for `pmt_mask`.

**Comment by a1exndr** [comment]
I'll delete and re-create this branch to avoid conflicts - I'm not an expert.

**Pull request closed without merging**